### PR TITLE
GH-127133: Remove ability to nest argument groups & mutually exclusive groups

### LINKED ISSUE #127133: Raise exception when nesting argument groups and mutually exclusive groups
The ability to nest argument groups and mutually exclusive groups was deprecated in 3.11 via https://github.com/python/cpython/pull/30098. 

At present, we raise warnings to let users know this has been deprecated. I'd like to remove the logic since we're now two (going on three) releases from initial deprecation, and doing so will also simplify the logic to support new features, like #55797.

### Linked PRs
* gh-127186

### PR DESCRIPTION
**Pull request by savannahostrowski**
This PR removes the ability to:
- Call `add_argument_group` on an argument group
- Call `add_argument_group` or `add_mutually_exclusive_group`  on a mutually exclusive group 

We have been warning users of this deprecation since 3.11. 

* Issue: gh-127133

----
📚 Documentation preview 📚: https://cpython-previews--127186.org.readthedocs.build/

### CODE REVIEW

File: Lib/argparse.py [hunk 1]
```diff
@@ -1709,14 +1709,7 @@ def _remove_action(self, action):
...
     def add_argument_group(self, *args, **kwargs):
-        import warnings
-        warnings.warn(
-            "Nesting argument groups is deprecated.",
-            category=DeprecationWarning,
-            stacklevel=2
-        )
-        return super().add_argument_group(*args, **kwargs)
-
+        raise ValueError('nested argument groups are not supported')
```
Comment (savannahostrowski) [inline Lib/argparse.py]: I *think* this is the right error type but let me know if there's a more appropriate type.

**Comment by savannahostrowski** [comment]
@iritkatriel Perhaps you can take a look at this one if you've got bandwidth, since you added the original deprecation warnings ☺️

**Comment by savannahostrowski** [comment]
Hmm, I think that there are likely other places where we might be able to simplify the code as a result of no longer supporting nested groups. Converting to draft.

File: Lib/test/test_argparse.py [hunk 2]
```diff
@@ -3297,6 +3304,14 @@ def test_empty_group(self):
         with self.assertRaises(ValueError):
             parser.parse_args(['-h'])
 
+    def test_nested_mutex_groups(self):
+        parser = argparse.ArgumentParser(prog='PROG')
+        g = parser.add_mutually_exclusive_group()
+        g.add_argument("--spam")
+        self.assertRaisesRegex(ValueError,
+                               'nested mutually exclusive groups are not supported',
+                                 g.add_mutually_exclusive_group)
```
Comment (iritkatriel) [inline Lib/test/test_argparse.py]:
```suggestion
                               'nested mutually exclusive groups are not supported',
                               g.add_mutually_exclusive_group)
```

File: Lib/argparse.py [hunk 3]
```diff
@@ -1737,15 +1730,8 @@ def _remove_action(self, action):
...
-        import warnings
-        warnings.warn(
-            "Nesting mutually exclusive groups is deprecated.",
-            category=DeprecationWarning,
-            stacklevel=2
-        )
-        return super().add_mutually_exclusive_group(*args, **kwargs)
-
+    def add_mutually_exclusive_group(self, **kwargs):
+        raise ValueError('nested mutually exclusive groups are not supported')
```
Comment (iritkatriel) [inline Lib/argparse.py]:
```suggestion
        raise ValueError('mutually exclusive groups cannot be nested')
```

File: Lib/argparse.py (hunk 1 above)
Comment (iritkatriel) [inline Lib/argparse.py]:
```suggestion
        raise ValueError('argument groups cannot be nested')
```

File: Doc/library/argparse.rst [hunk 4]
```diff
@@ -1926,11 +1926,10 @@ Argument groups
    Note that any arguments not in your user-defined groups will end up back
    in the usual "positional arguments" and "optional arguments" sections.
 
-   .. versionchanged:: 3.11
-    Calling :meth:`add_argument_group` on an argument group is deprecated.
-    This feature was never supported and does not always work correctly.
-    The function exists on the API by accident through inheritance and
-    will be removed in the future.
+   .. deprecated-removed:: 3.11 3.14
+      Calling :meth:`add_argument_group` on an argument group has been removed.
```
Comment (iritkatriel) [inline Doc/library/argparse.rst]: it hasn't been removed, it raises an exception.

**Review by iritkatriel — APPROVED** [review]

**Pull request merged**